src/data/file.ts

```diff
@@ -481,6 +486,20 @@ export function parsePage(file: TFile, cache: MetadataCache, markdownData: Parse
             let frontFields = parseFrontmatter(fileCache.frontmatter) as Record<string, LiteralValue>;
             for (let [key, value] of Object.entries(frontFields)) fields.set(key, value);
         }
+
+        // match up tasks by line number to assign their block IDs
+        if (fileCache.listItems && fileCache.listItems.length > 0 && markdownData.tasks.length > 0) {
+            let items = fileCache.listItems.reduce((out, t) => {
+                out[t.position.start.line] = t;
+                return out;
+            }, {} as Record<number, ListItemCache>);
+            markdownData.tasks.forEach(t => {
+                let block = items[t.line];
+                if (block) {
```

Comment:
Hm, this is odd - I was taking a look at linking in the UI and realized the blockIds aren't set correctly. Block objects look like `{parent: -7, position: {}, task: " "}`. Should I be using a different source for the blocks, @blacksmithgu?

Reply:
I believe there is a BlockCache you can use instead. I'm not sure if you can link to individual lines - you may need to link to the whole *list* (by doing a range check to see if the current line is in the block).